Another failure:

```
=========================== short test summary info ============================
FAILED tests/test_vm.py::test_vm_migration - assert String did not meet the expectations.
  
  🟢=EXPECTED | ⚪️=OPTIONAL | 🟡=UNEXPECTED | 🔴=REFUSED/UNMATCHED
  
  Here is the string that was tested: 
  
  🟢 outmigrate      | /nix/store/lpzzmnf3nwfjnwwym9kdnk4zgfkin22x-python3.11-fc.qemu-1.7dev/bin/fc-qemu -v outmigrate simplevm
  🟢 outmigrate      | load-system-config
  ⚪️ outmigrate      | simplevm      libceph ceph
  ⚪️ debug           | simplevm> ceph -c /etc/ceph/ceph.conf --name client.host1 --format json osd lspools
  ⚪️ outmigrate      | simplevm      libceph ceph                           output_line='\n'
  ⚪️ outmigrate      | simplevm      libceph ceph                           output_line='[{"poolnum":1,"poolname":"rbd"},{"poolnum":2,"poolname":"rbd.ssd"},{"poolnum":3,"poolname":"rbd.hdd"}]\n'
  ⚪️ outmigrate      | simplevm      libceph ceph                           returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.root
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.root","id":"1b849b6a9795","size":2147483648,"objects":512,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1b849b6a9795","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:26 2025","access_timestamp":"Tue Nov 11 21:41:26 2025","modify_timestamp":"Tue Nov 11 21:41:26 2025","parent":{"pool":"rbd.hdd","pool_namespace":"","image":"fc-21.05-dev","id":"103a64a9c6aa","snapshot":"v1","trash":false,"overlap":524288000}}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.swap
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.swap","id":"1ba1ddee10c7","size":1073741824,"objects":256,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1ba1ddee10c7","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:30 2025","access_timestamp":"Tue Nov 11 21:41:30 2025","modify_timestamp":"Tue Nov 11 21:41:30 2025"}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.tmp
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.tmp","id":"1bba245cc52b","size":5368709120,"objects":1280,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1bba245cc52b","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:30 2025","access_timestamp":"Tue Nov 11 21:41:30 2025","modify_timestamp":"Tue Nov 11 21:41:30 2025"}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.cidata
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.cidata","id":"1bd3355cc4ba","size":10485760,"objects":3,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1bd3355cc4ba","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:33 2025","access_timestamp":"Tue Nov 11 21:41:33 2025","modify_timestamp":"Tue Nov 11 21:41:33 2025"}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm              acquire-lock                   target='/run/qemu.simplevm.lock'
  🟢 outmigrate      | simplevm              acquire-lock                   count=1 result='locked' target='/run/qemu.simplevm.lock'
  🟢 outmigrate      | simplevm     qemu/qmp qmp_capabilities               arguments={} id=None
  🟢 outmigrate      | simplevm     qemu/qmp query-status                   arguments={} id=None
  🟢 outmigrate      | simplevm              outmigrate
  🟢 outmigrate      | simplevm              consul-register
  🟢 heartbeat       | simplevm              heartbeat-initialized
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.root
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.root","id":"1b849b6a9795","size":2147483648,"objects":512,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1b849b6a9795","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:26 2025","access_timestamp":"Tue Nov 11 21:41:26 2025","modify_timestamp":"Tue Nov 11 21:41:26 2025","parent":{"pool":"rbd.hdd","pool_namespace":"","image":"fc-21.05-dev","id":"103a64a9c6aa","snapshot":"v1","trash":false,"overlap":524288000}}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.root
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7053","address":"192.168.4.6:0/1386961170"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.swap
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.swap","id":"1ba1ddee10c7","size":1073741824,"objects":256,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1ba1ddee10c7","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:30 2025","access_timestamp":"Tue Nov 11 21:41:30 2025","modify_timestamp":"Tue Nov 11 21:41:30 2025"}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.swap
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7081","address":"192.168.4.6:0/2907907767"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.tmp
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.tmp","id":"1bba245cc52b","size":5368709120,"objects":1280,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1bba245cc52b","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:30 2025","access_timestamp":"Tue Nov 11 21:41:30 2025","modify_timestamp":"Tue Nov 11 21:41:30 2025"}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.tmp
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7106","address":"192.168.4.6:0/1623972147"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm              setup-outgoing-migration       cookie='32b232d8a87a76a8bab079f3da14d13866277a6f'
  🟢 outmigrate      | simplevm              locate-inmigration-service
  🟢 outmigrate      | simplevm              check-staging-config           result='none'
  ⚪️ outmigrate      | simplevm              waiting                        interval=3 remaining=3599
  ⚪️ outmigrate      | simplevm              check-staging-config           result='none'
  🟢 outmigrate      | simplevm              located-inmigration-service    url='http://host2.sto.test.gocept.net:37809'
  🟢 heartbeat       | simplevm              started-heartbeat-ping
  🟢 outmigrate      | simplevm              acquire-migration-locks
  🟢 heartbeat       | simplevm              heartbeat-ping
  🟢 outmigrate      | simplevm              check-staging-config           result='none'
  🟢 outmigrate      | simplevm         qemu acquire-migration-lock         result='success'
  🟢 outmigrate      | simplevm              acquire-local-migration-lock   result='success'
  🟢 outmigrate      | simplevm              acquire-remote-migration-lock
  🟢 outmigrate      | simplevm              acquire-remote-migration-lock  result='success'
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.root
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7053","address":"192.168.4.6:0/1386961170"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm         ceph unlock                         volume='rbd.ssd/simplevm.root'
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.root
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7053","address":"192.168.4.6:0/1386961170"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 lock rm rbd.ssd/simplevm.root host1 client.7053
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.swap
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7081","address":"192.168.4.6:0/2907907767"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm         ceph unlock                         volume='rbd.ssd/simplevm.swap'
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.swap
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7081","address":"192.168.4.6:0/2907907767"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 lock rm rbd.ssd/simplevm.swap host1 client.7081
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.tmp
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7106","address":"192.168.4.6:0/1623972147"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm         ceph unlock                         volume='rbd.ssd/simplevm.tmp'
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.tmp
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7106","address":"192.168.4.6:0/1623972147"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 lock rm rbd.ssd/simplevm.tmp host1 client.7106
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json info rbd.ssd/simplevm.cidata
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='{"name":"simplevm.cidata","id":"1bd3355cc4ba","size":10485760,"objects":3,"order":22,"object_size":4194304,"snapshot_count":0,"block_name_prefix":"rbd_data.1bd3355cc4ba","format":2,"features":["layering"],"op_features":[],"flags":[],"create_timestamp":"Tue Nov 11 21:41:33 2025","access_timestamp":"Tue Nov 11 21:41:33 2025","modify_timestamp":"Tue Nov 11 21:41:33 2025"}\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.cidata
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7131","address":"192.168.4.6:0/3637946395"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm         ceph unlock                         volume='rbd.ssd/simplevm.cidata'
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 --format json lock list rbd.ssd/simplevm.cidata
  ⚪️ outmigrate      | simplevm      libceph rbd                            output_line='[{"id":"host1","locker":"client.7131","address":"192.168.4.6:0/3637946395"}]\n'
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  ⚪️ outmigrate      | simplevm      libceph rbd
  ⚪️ debug           | simplevm> rbd -c /etc/ceph/ceph.conf --name client.host1 lock rm rbd.ssd/simplevm.cidata host1 client.7131
  ⚪️ outmigrate      | simplevm      libceph rbd                            returncode=0
  🟢 outmigrate      | simplevm              prepare-remote-environment
  🟢 outmigrate      | simplevm              start-migration                target='tcp:192.168.4.7:2345'
  🟢 outmigrate      | simplevm         qemu migrate
  🟢 outmigrate      | simplevm     qemu/qmp migrate-set-capabilities       arguments={'capabilities': [{'capability': 'xbzrle', 'state': False}, {'capability': 'auto-converge', 'state': True}]} id=None
  🟢 outmigrate      | simplevm     qemu/qmp migrate-set-parameters         arguments={'compress-level': 0, 'downtime-limit': 4000, 'max-bandwidth': 22500} id=None
  🟢 outmigrate      | simplevm     qemu/qmp migrate                        arguments={'uri': 'tcp:192.168.4.7:2345'} id=None
  🟢 outmigrate      | simplevm     qemu/qmp query-migrate-parameters       arguments={} id=None
  🟢 outmigrate      | simplevm         qemu migrate-parameters             announce-initial=50 announce-max=550 announce-rounds=5 announce-step=100 block-incremental=False compress-level=0 compress-threads=8 compress-wait-thread=True cpu-throttle-increment=10 cpu-throttle-initial=20 cpu-throttle-tailslow=False decompress-threads=2 downtime-limit=4000 max-bandwidth=22500 max-cpu-throttle=99 max-postcopy-bandwidth=0 multifd-channels=2 multifd-compression='none' multifd-zlib-level=1 multifd-zstd-level=1 throttle-trigger-threshold=50 tls-authz='' tls-creds='' tls-hostname='' x-checkpoint-delay=20000 xbzrle-cache-size=67108864
  🟢 outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  🟢 outmigrate      | simplevm              migration-status               mbps=0 remaining='286,334,976' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 0,
  ⚪️ debug           | simplevm>          'mbps': 0,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 0,
  ⚪️ debug           | simplevm>          'normal-bytes': 0,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 0,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 286334976,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 0},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 3}
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.32976 remaining='285,523,968' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 183,
  ⚪️ debug           | simplevm>          'mbps': 0.32976,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 15,
  ⚪️ debug           | simplevm>          'normal-bytes': 61440,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 10,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 285523968,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 63326},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 1419}
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.3264950495049505 remaining='285,327,360' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 211,
  ⚪️ debug           | simplevm>          'mbps': 0.3264950495049505,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 35,
  ⚪️ debug           | simplevm>          'normal-bytes': 143360,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 9,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 285327360,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 145818},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 3421}
  ⚪️ heartbeat       | simplevm              heartbeat-ping
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.18144 remaining='266,854,400' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 4711,
  ⚪️ debug           | simplevm>          'mbps': 0.18144,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 45,
  ⚪️ debug           | simplevm>          'normal-bytes': 184320,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 2500,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 266854400,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 227582},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 6251}
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.18144 remaining='225,894,400' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 14711,
  ⚪️ debug           | simplevm>          'mbps': 0.18144,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 45,
  ⚪️ debug           | simplevm>          'normal-bytes': 184320,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 2500,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 225894400,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 317902},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 10252}
  ⚪️ heartbeat       | simplevm              heartbeat-ping
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.18144 remaining='168,550,400' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 28711,
  ⚪️ debug           | simplevm>          'mbps': 0.18144,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 45,
  ⚪️ debug           | simplevm>          'normal-bytes': 184320,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 2500,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 168550400,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 444350},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 15911}
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.18144 remaining='86,630,400' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 48711,
  ⚪️ debug           | simplevm>          'mbps': 0.18144,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 45,
  ⚪️ debug           | simplevm>          'normal-bytes': 184320,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 2500,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 86630400,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 624990},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 23912}
  ⚪️ heartbeat       | simplevm              heartbeat-ping
  ⚪️ heartbeat       | simplevm              heartbeat-ping
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.32976 remaining='18,075,648' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 65401,
  ⚪️ debug           | simplevm>          'mbps': 0.32976,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 92,
  ⚪️ debug           | simplevm>          'normal-bytes': 376832,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 10,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 18075648,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 968992},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 35227}
  ⚪️ heartbeat       | simplevm              heartbeat-ping
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  ⚪️ outmigrate      | simplevm              migration-status               mbps=0.3264950495049505 remaining='634,880' status='active'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'expected-downtime': 4000,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 1,
  ⚪️ debug           | simplevm>          'duplicate': 69562,
  ⚪️ debug           | simplevm>          'mbps': 0.3264950495049505,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 189,
  ⚪️ debug           | simplevm>          'normal-bytes': 774144,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 9,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 634880,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 1405516},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'active',
  ⚪️ debug           | simplevm>  'total-time': 46542}
  ⚪️ heartbeat       | simplevm              heartbeat-ping
  ⚪️ outmigrate      | simplevm     qemu/qmp query-migrate                  arguments={} id=None
  🟢 outmigrate      | simplevm              migration-status               mbps=0.33345047728140514 remaining='0' status='completed'
  ⚪️ debug           | simplevm> {'blocked': False,
  ⚪️ debug           | simplevm>  'downtime': 9,
  ⚪️ debug           | simplevm>  'ram': {'dirty-pages-rate': 0,
  ⚪️ debug           | simplevm>          'dirty-sync-count': 3,
  ⚪️ debug           | simplevm>          'duplicate': 69648,
  ⚪️ debug           | simplevm>          'mbps': 0.33345047728140514,
  ⚪️ debug           | simplevm>          'multifd-bytes': 0,
  ⚪️ debug           | simplevm>          'normal': 259,
  ⚪️ debug           | simplevm>          'normal-bytes': 1060864,
  ⚪️ debug           | simplevm>          'page-size': 4096,
  ⚪️ debug           | simplevm>          'pages-per-second': 564,
  ⚪️ debug           | simplevm>          'postcopy-requests': 0,
  ⚪️ debug           | simplevm>          'remaining': 0,
  ⚪️ debug           | simplevm>          'skipped': 0,
  ⚪️ debug           | simplevm>          'total': 286334976,
  ⚪️ debug           | simplevm>          'transferred': 1694150},
  ⚪️ debug           | simplevm>  'setup-time': 2,
  ⚪️ debug           | simplevm>  'status': 'completed',
  ⚪️ debug           | simplevm>  'total-time': 52382}
  🟢 outmigrate      | simplevm     qemu/qmp query-status                   arguments={} id=None
  🟢 outmigrate      | simplevm              finish-migration
  ⚪️ outmigrate      | simplevm         qemu vm-destroy-kill-supervisor     attempt=1
  ⚪️ outmigrate      | simplevm         qemu vm-destroy-kill-supervisor     attempt=2
  🟢 outmigrate      | simplevm         qemu vm-destroy-vm-via-qmp
  🟢 outmigrate      | simplevm     qemu/qmp quit                           arguments={} id=None
  🟢 outmigrate      | simplevm         qemu clean-run-files
  🟢 outmigrate      | simplevm              finish-remote
  🟢 outmigrate      | simplevm              consul-deregister
  🟢 outmigrate      | simplevm              outmigrate-finished            exitcode=0
  🟢 outmigrate      | simplevm              release-lock                   count=0 target='/run/qemu.simplevm.lock'
  🟢 outmigrate      | simplevm              release-lock                   result='unlocked' target='/run/qemu.simplevm.lock'
  🟡                 | main-exit␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠exitcode=0
================== 1 failed, 107 passed in 779.69s (0:12:59) ===================
```

--- a/changelog.d/20251111_214431_fc-25.05-dev_scriv.md
+++ b/changelog.d/20251111_214431_fc-25.05-dev_scriv.md
@@ -1,0 +1,29 @@
+<!--
+
+A new changelog entry.
+
+Delete placeholder items that do not apply. Empty sections will be removed
+automatically during release.
+
+Leave the XX.XX as is: this is a placeholder and will be automatically filled
+correctly during the release and helps when backporting over multiple platform
+branches.
+
+-->
+
+### Impact
+
+<!-- Impact means "when this change is rolled out, there
+     might be interruptions/downtimes/required actions/... that
+     IMPACT THE RUNNING APPLICATION NEGATIVELY.
+
+     Having new features or changed is not an "impact". That's what
+     the main changelog (see below) is for.
+     -->
+
+- A bullet item for the Impact category.
+
+
+### NixOS XX.XX platform
+
+- Multiple bugfixes in fc.qemu for edge cases improving resiliency.

--- a/pkgs/fc/default.nix
+++ b/pkgs/fc/default.nix
@@ -54,8 +54,8 @@ rec {
       repo = "fc.qemu";
       # The release tooling didn't upgrade properly so we had to pick a specific
       # commit instead.
-      rev = "e0a1578849fe2f3261b38928fa41f06956ab511d";
-      hash = "sha256-QOFjdD62IjdS/5kYR7R1HMZiT23OTojnR+mrbCribo4=";
+      rev = "0bd0bf02345cbb9caf1d96f95fb6fb8b1356b771";
+      hash = "sha256-LAGvvfZZQdJN40QciiGuTLISRknpcokmZUMsYFc97Hk=";
     };
     fc-ceph = ceph;
     qemu_ceph = pkgs.qemu-ceph-nautilus;

